overlord/servicestate/quota_control.go

```diff
@@ -41,7 +41,21 @@ func checkSystemdVersion() {
 }
 
 func init() {
+	EnsureQuotaUsability()
+}
+
+// EnsureQuotaUsability is exported for unit tests from other packages to re-run
+// the init() time checks for quota usability which set the errors which
+// quotaGroupsAvailable() checks for.
+// It saves the previous state of the usability errors to be restored via the
+// provided restore function.
+func EnsureQuotaUsability() (restore func()) {
```

Comment:
cc @atesburak and @Meulengracht this is the direction I think we should take to support checking for whether a given cgroup is enabled, etc. as it allows us to most easily mock the needed things from other packages for their unit tests such as the manager test I wrote below here

Reply:
I completely agree here.

maybe nitpick: but I would prefer to see ensureQuotaUsability not being exported directly, and instead maybe being exported like we do other places where have adjacent test files like export_test.go (which lives in servicestate as well) which wraps the function with the restore functionality. It seems off to have this in the middle of the implementation file.

Reply:
The issue with this is that if we don't export it, then we can't use it from the overlord/managers_test.go because that is in a separate package. It is a terrible travesty in Go that you can define something in `pkg/export_test.go` and access the exported thing from export_test.go in `pkg/foo_test.go` but you CANNOT access that same thing in `pkg2/foo2_test.go` so hence why it is exported

Reply:
I didn't mean not to export it, I meant to move the entire EnsureUsabilityQuota to export_test.go and then in init call checkSystemdVersion() instead. This would seperate the unit test functionality from the actual implementation. But I don't feel that strongly about it, and its just more a nitpicky comment. Feel free to resolve this if you don't agree, I have approved your PR!

Reply:
if EnsureUsabilityQuota was moved to export_test.go it still wouldn't be importable from other packages though